cmd/ttn-lw-cli/internal/util/flags.go

```diff
@@ -466,14 +478,23 @@ func SetFields(dst interface{}, flags *pflag.FlagSet, prefix ...string) error {
 	return nil
 }
 
-var textUnmarshalerType = reflect.TypeOf((*encoding.TextUnmarshaler)(nil)).Elem()
+var (
+	errFieldRequired    = errors.DefineInvalidArgument("field_is_required", "field `{field}` is required")
+	textUnmarshalerType = reflect.TypeOf((*encoding.TextUnmarshaler)(nil)).Elem()
+)
 
 func setField(rv reflect.Value, path []string, v reflect.Value) error {
 	rt := rv.Type()
 	vt := v.Type()
 	props := proto.GetProperties(rt)
 	for _, prop := range props.Prop {
 		if prop.OrigName == path[0] {
+			if len(path) == 1 && v.String() == "null" {
```

Comment:
This should take the `field` type into account as well. Move this statement inside the `if` on `L499` and perhaps set the field to `nil` value explicitly?

Reply:
Would we get anything from setting the field to nil instead of not setting it at all? Preferred the latter.

Reply:
Maybe @htdvisser can comment on correctness here(e.g. if the original value may be non-zero, e.g a default value, I could imagine `--otaa` flag on create could set some fields, which the user may want to override), but it does seem like we always write for each other field type, so good to do at least for consistency's sake.

Reply:
Added an explicit `--unset` option, this discussion is no longer relevant.